### ISSUE TITLE
How to measure performances of GraphQL queries / database operations?

### ISSUE THREAD

**Issue by GP4cK** [doug-martin/nestjs-query#931]
I'd like to understand why a specific query is taking a couple seconds to resolve.
I saw from [web.dev](https://web.dev/custom-metrics/?utm_source=devtools#server-timing-api) that you could include a Server-Timing header with how much time was taken by the database, by the api etc.
I also saw that there is this little package [server-timing](https://www.npmjs.com/package/server-timing) that you can use to add these metrics to your response.

I guess I could use an [interceptor](https://docs.nestjs.com/interceptors) to measure how long in total the request takes but how to get more details on the database operations?

**Comment by hrmcdonald** [comment]
Under the hood, GraphQL in NestJS is actually just an Apollo server. So you should be able to enable [Apollo Tracing](https://www.apollographql.com/blog/exposing-trace-data-for-your-graphql-server-with-apollo-tracing-97c5dd391385/). You should be able to do this by just passing `tracing: true` in your NestJS GraphQL module `forRoot()` import config. Other than that you'd have to look into some sort of APM solution.

**Comment by Gujie-Novade** [comment]
Thanks @hrmcdonald. I'll try that.
@smolinari I'm using Postgres.

**Comment by GP4cK** [comment]
Thanks @hrmcdonald, I'll try that.
@smolinari I'm using postgres. Actually, I was able to see the duration of my db queries on Azure using Application Insights, and they're all super fast.
I managed to profile my app and it seems the app is taking breaks between function executions but I don't have an explanation for it.
I've asked a [question](https://stackoverflow.com/questions/66674837/profiling-my-nodejs-server-shows-its-hanging-but-how-to-figure-out-why) on StackOverflow but I think we can close this.
